bgpd/bgp_route.c

```diff
@@ -1283,8 +1281,9 @@ int bgp_path_info_cmp(struct bgp *bgp, struct bgp_path_info *new,
 			newl = new->extra->num_labels;
 		if (exist->extra)
 			existl = exist->extra->num_labels;
-		if (((new->extra &&bgp_is_valid_label(&new->extra->label[0])) !=
-		     (exist->extra &&
+		if (((new->extra && newl && bgp_is_valid_label(
```

Comment:
Can we make this condition readable? At least assign two new variables to compare.

Reply:
done